src/reactive/RouterSubject.js

```diff
@@ -0,0 +1,32 @@
+// notice_start
+/*
+ * Copyright 2015 Dev Shop Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// notice_end
+
+import Subject from '../../src/reactive/Subject';
+import RouterObservable from './RouterObservable';
+
+export default class RouterSubject extends Subject {
+    constructor(router) {
+        super();
+        this._router = router;
+    }
+    asRouterObservable() {
+        var source = this;
+        var subscribe =  observer => source.subscribe(observer);
```

Comment:
no. This is actually a factory to invoke the subscribe not the actually subscription. the Observable objects and ultimately the Subject take a subscribe method via the ctor.